### PR TITLE
Add drag-drop list attributes

### PR DESCRIPTION
**Pull request by tacettinkustu**
In this project, I added some functionality to my application to make it interactive. The user be able to reorder the list by dragging each item and moving it into the desired position. The user also be able to mark task completion by selecting the corresponding checkbox (or undo it by unchecking the checkbox). The updated tasks list is stored in local storage. For this purpose, I followed these steps:

1. Add 2 new JavaScript files and import them as modules.
2. Add event listeners to list items (dragstart, dragover, drop) and to checkbox (change).
3. Update items object's values for index and completed keys upon user actions.
4. Store the updated array of items in local storage, so the user gets the correct list values after the page reloads.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -20,7 +20,8 @@
   },
   "homepage": "https://github.com/tacettinkustu/microverse-todo-list#readme",
   "dependencies": {
-    "lodash": "^4.17.21"
+    "lodash": "^4.17.21",
+    "sortablejs": "^1.14.0"
   },
   "devDependencies": {
     "css-loader": "^5.2.7",
```
Comment (carlos-ssh) [inline package.json:27]: - After run `npm i`  and  type `npm run build` tried to run by multiple ways locally your project and I head an issue with your **webpack setup file.** Kindly work on it.
Screen Shot 2021-07-15 at 03 44 30

**Review by carlos-ssh — CHANGES_REQUESTED** [review]
Hi @tacettinkustu, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- x
- y
- z

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: package.json (hunk 1 above)
Comment (tacettinkustu) [reply package.json:27]: I change some steps for setting up environment.

File: src/assets/js/sorting.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        const order = localStorage.getItem(sortable.options.group.name);
+        return order ? order.split('|') : [];
+      },
+
+      /**
+       * Save the order of elements. Called onEnd (when the item is dropped).
+       * @param {Sortable}  sortable
+       */
+      set(sortable) {
+        const order = sortable.toArray();
```
Comment (DeliceLydia) [inline src/assets/js/sorting.js:24]: - [ ] I can see that you added sorting method but as the requirements mentioned you need to add drag / drop methods in the same file as well
- [ ] Another file should contain methods related to the status updates (`completed`: `true` / `false`).
- [ ] You are supposed to import these files as modules

File: src/index.html [hunk 3]
```diff
@@ -14,16 +14,15 @@
...
-        </li>
-        <li class="input d-flex-row">
+        </div>
+        <div class="input d-flex-row">
           <input placeholder="Add to your list..." type="text" />
           <i class="fas fa-level-down-alt"></i>
-        </li>
-      </ul>
+        </div>
+        <ul class="list" id="items"></ul>
```
Comment (DeliceLydia) [inline src/index.html:25]: - [ ] The project is supposed to have an event listeners on the list items (`dragstart`, `dragover`, `drop`) and to checkbox (`change`).
- [ ] Currently when I click on the checkbox, I can't see any change in the local storage that `completed` changed from `false` to `true` or vice verse. Please work on that

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
## Hi Team, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
_Check the comments under the review._


### Cheers and Happy coding!👏👏👏

Feel free to leave a question, or a comment on the PR thread if something is not clear 100%. 

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/assets/js/sorting.js (hunk 2 above)
Comment (tacettinkustu) [reply src/assets/js/sorting.js]: I used drag/drop methods. Updated completed condition of todos. And import files as modules.

File: src/index.html (hunk 3 above)
Comment (tacettinkustu) [reply src/index.html]: I used event listeners on the list items (dragstart, dragover, drop) and to checkbox (change).

**Review by sinansevgi — APPROVED** [review]
# Approved :heavy_check_mark: 

Hi @tacettinkustu , 

Great work with making all the changes requested by previous Tses.
:+1: Drag and drop works perfectly.

Your project is complete!  
Congratulations! 🎉

You can merge this pull request and continue with your journey :wave: 


Happy Hacking! :computer: 

-----

You can reply to this conversation if you have any questions or need elaboration. :wave: 
Sinan Sevgi


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**